### PR TITLE
[PWA-2498] Rename Link's prefetchType prop

### LINKED ISSUE #3647: [Issue] Rename Link's prefetchType prop
This issue is automatically created based on existing pull request: magento/pwa-studio#3646: Rename Link's prefetchType prop
## 
## Description

For the `Link` component, rename the boolean prop `prefetchType` to `shouldPrefetch`. The former appears to indicate it will contain a type rather than a boolean, causing some confusion. The approach herein

- Deprecates the existing `prefetchType` prop to `Link` and `useLink`, but does not remove it
- Adds `shouldPrefetch` as a new prop to both component and talon
- Ignores the value of `prefetchType` unless it is `true`

## Related Issue

#3308 

### Verification Stakeholders

- @jcalcaben 
- @luke-denton-aligent 

## Test Plan
Note - Link component not being used on Venia 
1. Unit tests should pass
2. Cypress Test should pass

## Breaking Changes (if any)

These changes should be non-breaking. Eventually, we should remove the deprecated prop, which would be a breaking change.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### PR DESCRIPTION
**Pull request by jimbo**
## Description

For the `Link` component, rename the boolean prop `prefetchType` to `shouldPrefetch`. The former appears to indicate it will contain a type rather than a boolean, causing some confusion. The approach herein

- Deprecates the existing `prefetchType` prop to `Link` and `useLink`, but does not remove it
- Adds `shouldPrefetch` as a new prop to both component and talon
- Ignores the value of `prefetchType` unless it is `true`

## Related Issue

Closes #3647

### Verification Stakeholders

- @jcalcaben 
- @luke-denton-aligent 

## Verification Steps

Links with either value of `prefetchType` should work the same as they do now. That said, we don't appear to have any.

## Breaking Changes (if any)

These changes should be non-breaking. Eventually, we should remove the deprecated prop, which would be a breaking change.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### Resolved issues:
1. [x] resolves magento/pwa-studio#3647: Rename Link's prefetchType prop

### CODE REVIEW

**Comment by supernova-at** [comment]
@magento create issue from PR

File: packages/venia-ui/lib/components/Link/link.js [hunk 1]
```diff
@@ -3,38 +3,40 @@ import { Link as RouterLink } from 'react-router-dom';
...
+ * @property {bool} shouldPrefetch `true` activates prefetching the target page when the link becomes visible in the viewport.
  * @property {string} to From [react-router-dom Link](https://knowbody.github.io/react-router-docs/api/Link.html). The absolute path to the target page of the link. Uses the `to` prop from the `react-router-dom Link`.
  *
  * @example <caption>Basic usage</caption>
- * <Link prefetchType={true} to="/about/">About Us</Link>
+ * <Link shouldPrefetch={true} to="/about/">About Us</Link>
  */
 const Link = props => {
-    const talonProps = useLink(props);
+    // TODO: remove `prefetchType`
```
Comment (jeremycharron) [inline packages/venia-ui/lib/components/Link/link.js:16]: Should this TODO be removed?

**Review by jeremycharron — COMMENTED** [review]
@jimbo 
Code looks good except single comment about TODO. I added a minor version label + fixed the "closes issue" in description. Warning should be gone next checks run.

File: packages/venia-ui/lib/components/Link/link.js (hunk 1 above)
Comment (jimbo) [reply packages/venia-ui/lib/components/Link/link.js:16]: No, the prop still exists so as not to be a breaking change. We can actually remove it in a future major release, and delete the comment at that time.

**Review by jeremycharron — APPROVED** [review]
Code looks good. Approved.

**Comment by ericeoeur** [comment]
QA has passed for this ticket and will be merged momentarily. Thanks!

**Pull request merged**